### PR TITLE
Display custom error message for NotImplementedErrors

### PR DESCRIPTION
**Pull request by giograno**
Implements #8053

### CODE REVIEW

**Review by lukqw — COMMENTED** [review]
When I tried to look into this issue I noticed that the `... yet implemented or pro feature ...` message is also printed in `localstack/aws/handlers/service.py` and `localstack/utils/aws/request_context.py`. 
Do we need to adjust the logic in there as well?

**Comment by steffyP** [comment]
> When I tried to look into this issue I noticed that the `... yet implemented or pro feature ...` message is also printed in `localstack/aws/handlers/service.py` and `localstack/utils/aws/request_context.py`.
> Do we need to adjust the logic in there as well?

I think it would make sense. [localstack/utils/aws/request_context.py](https://github.com/localstack/localstack/blob/master/localstack/utils/aws/request_context.py#L175) seems to patch moto responses -> moto sometimes adds additional information when a specific parameters or actions are not implemented. 

Not quite sure when we would hit [localstack/aws/handlers/service.py](https://github.com/localstack/localstack/blob/master/localstack/aws/handlers/service.py#L185), I guess there are other edge-cases.

**Comment by thrau** [comment]
Hey folks! The code that's being modified here is code that's actually not used in the handler chain. The handler chain has its own dispatching mechanism and also its own exception serializer, which happens here:

https://github.com/localstack/localstack/blob/e1b80232a5aa08f5ccc2b3f7c0950297bbe498f1/localstack/aws/handlers/service.py#L180-L193

Sorry about the confusion, I can see this may needs some consolidation

File: tests/unit/aws/test_skeleton.py [hunk 1]
```diff
@@ -186,9 +203,23 @@ def test_skeleton_e2e_sqs_send_message():
...
 
 
-def test_skeleton_e2e_sqs_send_message_not_implemented():
+@pytest.mark.parametrize(
+    "api_class, oracle_message",
+    [
+        (
+            TestSqsApiNotImplemented(),
+            "API action 'SendMessage' for service 'sqs' not yet implemented or pro feature"
+            " - check https://docs.localstack.cloud/user-guide/aws/feature-coverage for further information",
```
Comment (giograno) [inline tests/unit/aws/test_skeleton.py:212]: This string should change accordingly with #8052.

**Review by steffyP — APPROVED** [review]
LGTM 👍 thanks for implementing 🙏

**Comment by lukqw** [comment]
@giograno I have merged latest master into this branch and adjusted the files accordingly 👍 
Feel free to merge as-is or make further adjustments to your liking.

**Pull request merged**